File: models/dvd.js
const mongoose = require('mongoose');
const EpisodeSchema = require('./episode_schema');
const BookmarkSchema = require('./bookmark_schema');
const TagSchema = require('./tag_schema');
const Schema = mongoose.Schema;

const DvdSchema = new Schema({
  title: { type: String, required: [true, 'Title is required.'], index: true, text: true },
  rating: Number,
  abstractTxt: String,
  abstractSource: String,
  abstractUrl: String,
  imageUrl: String,
  fileUrl: String,
  playbackTime: Number,
  episodes: [EpisodeSchema],
  bookmarks: [BookmarkSchema],
  tags: [TagSchema]
},
{
  timestamps: true
});

const Dvd = mongoose.model('dvd', DvdSchema);
module.exports = Dvd;

File: models/dvd.go
package models

import (
	"github.com/jinzhu/gorm"
)

type Dvd struct {
	gorm.Model
	Title string
	Rating int
	AbstractTxt string
	AbstractSrc	string
	AbstractURL	string
	ImageURL string
	FileURL	string
	PlaybackTime int
	Episodes []Episode
	Bookmarks []Bookmark
	Tags []Tag `gorm:"many2many:dvd_tags;"`
}
